core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -447,6 +468,47 @@ bool core_mmu_find_table(vaddr_t va, unsigned max_level,
 	return true;
 }
 
+bool core_mmu_divide_block(struct core_mmu_table_info *tbl_info,
+			   unsigned int idx)
+{
+	uint32_t *new_table;
+	uint32_t *entry;
+	uint32_t new_table_desc;
+	paddr_t paddr;
+	uint32_t attr;
+	int i;
+
+	if (tbl_info->level != 1)
+		return false;
+
+	if (idx >= NUM_L1_ENTRIES)
+		return false;
+
+	new_table = core_mmu_alloc_l2(NUM_L2_ENTRIES * SMALL_PAGE_SIZE);
+	if (!new_table)
+		return false;
+
+	entry = (uint32_t *)tbl_info->table + idx;
+	assert(get_desc_type(1, entry) == DESC_TYPE_SECTION ||
+	       get_desc_type(1, entry) == DESC_TYPE_SUPER_SECTION);
```

Comment:
If it is a `DESC_TYPE_SUPER_SECTION` then the surrounding `DESC_TYPE_SUPER_SECTION` entries need to be converted into `DESC_TYPE_SECTION`. But that's overkill as we don't use `DESC_TYPE_SUPER_SECTION` in OP-TEE for the moment. Asserting only `DESC_TYPE_SECTION` would be more correct.